### PR TITLE
Feature vsp fuel tanks

### PR DESCRIPTION
**Pull request by timdmacdo**
This allows fuel tanks to be specified and volumes to be determined in OpenVSP. Fuel tank available mass and cg can also be specified manually. In both cases a possible cg range can be computed.

### CODE REVIEW

File: trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+    Properties Used:
+    N/A
+    """       
+    fuel_tanks = Data()
+    
+    if 'wings' in vehicle:
+        for wing in vehicle.wings:
+            if 'Fuel_Tanks' in wing:
+                for tank in wing.Fuel_Tanks:
```
Comment (WallyMaier) [inline trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py]: loop-ception

File: trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
+## @ingroup Input_Output-OpenVSP
```
Comment (planes) [inline trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py]: rename this. The name props is unclear. Props could be propulsion, propeller, etc...

File: trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    """      
+    
+    # Reset OpenVSP to avoid including a previous vehicle
+    vsp.ClearVSPModel()    
+    vsp.ReadVSPFile(tag + '.vsp3')
+    
+    # Extract fuel tanks from vehicle
+    fuel_tanks = get_fuel_tank_tags(vehicle)
+    
+    num_slices = 100 # Slices used to estimate mass distribution from areas in OpenVSP
```
Comment (planes) [inline trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py]: default this in some way

File: trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    vehicle.wings.*.Fuel_Tanks.*.mass_properties.
+      center_of_gravity                      [m]
+      full_fuel_mass                         [kg]
+      full_fuel_volume                       [m^3]
+      
+
+    Properties Used:
+    N/A
+    """      
+    if 'wings' in vehicle:
```
Comment (planes) [inline trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py]: This if is unnecessary, all SUAVE vehicles have wings data:
https://github.com/suavecode/SUAVE/blob/develop/trunk/SUAVE/Vehicle.py#L52

File: trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        for wing in vehicle.wings:
+            if 'Fuel_Tanks' in wing:
```
Comment (planes) [inline trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py]: Looks like Fuel_Tanks are built into wings by you adding the container.

File: trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+                for tank in wing.Fuel_Tanks:
+                    tank.mass_properties.center_of_gravity = fuel_tanks[tank.tag].center_of_gravity
+                    tank.mass_properties.full_fuel_mass    = fuel_tanks[tank.tag].full_fuel_mass
+                    tank.mass_properties.full_fuel_volume  = fuel_tanks[tank.tag].volume
+                    
+    if 'fuselages' in vehicle:
```
Comment (planes) [inline trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py]: Same, fuselage is inherent in vehicles.

File: trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+    Outputs:    
+    fuel_tanks.tag                           [-]
+
+    Properties Used:
+    N/A
+    """       
+    fuel_tanks = Data()
+    
+    if 'wings' in vehicle:
```
Comment (planes) [inline trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py]: we don't need ifs

File: trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+        for wing in vehicle.wings:
+            if 'Fuel_Tanks' in wing:
+                for tank in wing.Fuel_Tanks:
+                    fuel_tanks[tank.tag] = Data()
+                    
+    if 'fuselages' in vehicle:
+        for fuse in vehicle.fuselages:
+            if 'Fuel_Tanks' in fuse:
+                for tank in fuse.Fuel_Tanks:
+                    fuel_tanks[tank.tag] = Data()
```
Comment (planes) [inline trunk/SUAVE/Input_Output/OpenVSP/get_fuel_tank_props.py]: is this the best way to store tags?

File: trunk/SUAVE/Methods/Center_of_Gravity/compute_possible_longitudinal_fuel_center_of_gravity.py [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+    min_cg                    [m]
+    max_cg                    [m]
+
+    Properties Used:
+    N/A
+    """  
+    
+    fuel_tanks = []
+    
+    if 'wings' in vehicle:
```
Comment (planes) [inline trunk/SUAVE/Methods/Center_of_Gravity/compute_possible_longitudinal_fuel_center_of_gravity.py]: more ifs

File: regression/scripts/concorde/concorde.py [hunk 10]
```diff
@@ -42,6 +42,10 @@
 
 # This is a sizing function to fill turbojet parameters
 from SUAVE.Methods.Propulsion.turbojet_sizing import turbojet_sizing
+from SUAVE.Methods.Center_of_Gravity.compute_possible_longitudinal_fuel_center_of_gravity \
```
Comment (planes) [inline regression/scripts/concorde/concorde.py]: I think we should consider a different name, usually we'd call this a CG range.

File: trunk/SUAVE/Components/Energy/Storages/Fuel_Tanks/Fuel_Tank.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+class Fuel_Tank(Energy_Component):
+    """
+    Energy Component object that stores fuel. Contains values
+    used to indicate its fuel type.
+    """
+    def __defaults__(self):
+        self.mass_properties.empty_mass            = 0.0
+        self.mass_properties.fuel_mass_when_full   = 0.0
+        self.mass_properties.fuel_volume_when_full = 0.0
+        self.fuel_type                  = None
```
Comment (planes) [inline trunk/SUAVE/Components/Energy/Storages/Fuel_Tanks/Fuel_Tank.py]: align

**Pull request merged**